octokit_utils.rb

```diff
@@ -1,235 +1,140 @@
-'puppetlabs-test_device',
- 'cisco-network-puppet-module',
- 'cisco_ios-private',
-'cisco_ios',
-
-  ]
-
-  SUPPORTED_MODULES_REGEX = "^((#{SUPPORTED_MODULES.join('|')})|modulesync_configs)$"
-
   def initialize(access_token)
     Octokit.auto_paginate = true
-    @client = Octokit::Client.new(:access_token => "#{access_token}")
+    @client = Octokit::Client.new(access_token: access_token.to_s)
     client.user.login
 
     @pr_cache = {}
   end
 
+  def load_module_list(modulefile)
+    output = File.read(modulefile)
+    parsed = JSON.parse(output)
+    parsed
+  end
+
   # Octokit uses a different method for getting the repositories of an
   # orginazation than it does for getting the repositories of a user. This
   # method checks the "type" of a given namespace and uses the value to
   # determine the method to call.
-  def ns_repos(namespace)
-    ns_type = (client.user "#{namespace}").type
-    if ns_type.eql? 'Organization'
-      client.organization_repositories(namespace)
-    else
-      client.repositories(namespace)
-    end
-  end
-
-  def list_repos(namespace, options)
-    if not options[:repo_regex]
-      regex = '.*'
-    else
-      regex = options[:repo_regex]
-    end
-
-    repos ||= ns_repos(namespace).collect {|ns_repo| ns_repo[:name] if ns_repo[:name] =~ /#{regex}/}
-    # The collection leaves nil entries in for non-matches
-    repos = repos.select {|repo| repo }
-    return repos.sort.uniq
-  end
 
   def pulls(repo, options)
     @pr_cache[[repo, options]] ||= client.pulls(repo, options)
   end
 
-  def fetch_async(repo, options={:state=>'open', :sort=>'updated'}, filter=[:statuses, :pull_request_commits, :issue_comments], limit=nil)
-  #example of limit..  limit={:attribute=>'closed_at', :date=>'2016-01-15 12:59:47 UTC'}
+  def fetch_async(repo, options = { state: 'open', sort: 'updated' }, filter = %i[statuses pull_request_commits issue_comments], limit = nil)
+    # example of limit..  limit={:attribute=>'closed_at', :date=>'2016-01-15 12:59:47 UTC'}
     pr_information_cache = []
     prs = []
     unlimited_prs = client.pulls(repo, options)
 
-    #there can be a large number of PRs returned, this limits the date range we are looking in
+    # there can be a large number of PRs returned, this limits the date range we are looking in
     if limit.nil?
       prs = unlimited_prs
     else
       unlimited_prs.each do |iter|
-        if limit[:attribute] == 'closed_at'
-          if iter.closed_at > limit[:date].to_time
-            prs.push(iter)
-          end
-        end
+        next unless limit[:attribute] == 'closed_at'
+
+        prs.push(iter) if iter.closed_at > limit[:date].to_time
       end
     end
 
     poolsize = 10
     mutex = Mutex.new
 
-    poolsize.times.map {
-      Thread.new(prs, pr_information_cache) do |prs, pr_information_cache|
-        while pr = mutex.synchronize { prs.pop }
+    poolsize.times.map do
+      Thread.new(prs, pr_information_cache) do |prs1, pr_information_cache1|
```

Comment:
Not entirely sure about the variable names `prs1` and `pr_information_cache1`. Looking at the scope of these vars, they shouldn't conflict with any other ones, unless I'm missing something?

Reply:
They don't , but here was a Rubocop offence : "Lint/ShadowingOuterLocalVariable: Shadowing outer local variable", I'll change the names of variables to be more suggestive

Reply:
I encountered this myself with Rubocop 😞